### ISSUE TITLE
Private pypi servers

### ISSUE THREAD

**Issue by greysteil** [dependabot/dependabot-core#389]
_From @khornberg on December 22, 2017 13:56_

Are there plans to enable private pypi servers?

_Copied from original issue: dependabot/feedback#65_

**Comment by greysteil** [comment]
Yes! I need to do a bit of work to enable it first, but should be able to ship something today or on Sunday.

**Comment by greysteil** [comment]
Private PyPI servers should be working now! Dependabot will read your `pip.conf` and/or your `requirements.txt` to find them.

Let me know if you have any trouble - it's a brand new features so may not be perfect, but I've tested internally.

🍸

**Comment by lithammer** [comment]
This is problematic if the private PyPi server isn't reachable outside a private network.

**Comment by greysteil** [comment]
How so? I mean, Dependabot won't be able to do dependency updates for you if you use a PyPI server it can't access, but the logic in Core isn't the problem there I don't think, and should work if you were self-hosting using Dependabot Script.

**Comment by lithammer** [comment]
Well we're mostly using a private PyPi mirror for performance and reliability reasons rather than the packages themselves. Our `pip.conf` is looking something like this and is checked into the repo(s).

```
[global]
index-url = https://pypi.example.com/simple
extra-index-url = https://pypi.org/simple
```

But since dependabot can't access pypi.example.com, it stopped. I mean, I suppose that makes sense. But I guess what I'm asking is something like any of these options:

- Read and use `extra-index-url` if `index-url` failed.
- Configuration to always try and check what it can against `https://pypi.org/simple`. Usually it's third-party libs you want automatically tracked rather than your own private packages.
- Option to ignore `pip.conf` (and similar files).

>[...] and should work if you were self-hosting using Dependabot Script.

Yes, I suppose it would.

I managed to work around the issue by adding `https://pypi.org/simple` as a config variable and checking the "Only use this index (instead of augmenting the base PyPI index)" box.

**Comment by greysteil** [comment]
> I managed to work around the issue by adding https://pypi.org/simple as a config variable and checking the "Only use this index (instead of augmenting the base PyPI index)" box.

Ooh, that is a perfect solution! I'm going to recommend that to anyone else who has this issue, if that makes sense to you?

**Comment by lithammer** [comment]
>Ooh, that is a perfect solution! I'm going to recommend that to anyone else who has this issue, if that makes sense to you?

Sorry, forgot to answer. Sure 🙂
